src/Chainweb/Miner/RestAPI.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- |
+-- Module: Chainweb.Miner.RestAPI
+-- Copyright: Copyright © 2019 Kadena LLC.
+-- License: MIT
+-- Maintainer: Colin Woodbury <colin@kadena.io>
+-- Stability: experimental
+--
+--
+module Chainweb.Miner.RestAPI
+  ( -- * Mining Work Submission
+    MiningSubmissionApi
+    -- * Mining Results
+  , MiningResultApi_
+  , MiningResultApi
+  , miningResultApi
+  , someMiningResultApi
+  ) where
+
+import Data.Proxy (Proxy(..))
+
+import Servant.API
+
+-- internal modules
+
+import Chainweb.Miner.Core (HeaderBytes, WorkBytes)
+import Chainweb.RestAPI.Utils (ChainwebEndpoint(..), Reassoc, SomeApi(..))
+import Chainweb.Version
+
+import Data.Singletons
+
+---
+
+-- -----------------------------------------------------------------------------
+-- Mining Work Submission
+
+-- | Shared between the `Chainweb.Miner.Miners.remoteMining` function and the
+-- /chainweb-miner/ executable.
+--
+type MiningSubmissionApi = "submit" :> ReqBody '[OctetStream] WorkBytes :> Post '[JSON] ()
```

Comment:
Note: this is used to call a `chainweb-miner`, and hence doesn't follow the usual Chainweb URL structure. It is defined here for optimal code sharing.